server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -63,6 +63,15 @@ public class ContextIndexSearcher extends IndexSearcher implements Releasable {
     private QueryProfiler profiler;
     private final MutableQueryTimeout cancellable;
 
+    // Constructor which doesn't wrap and doesn't throw an exception.
+    public ContextIndexSearcher(IndexReader reader, Similarity similarity, QueryCache queryCache, QueryCachingPolicy queryCachingPolicy) {
```

Comment:
Why is this new constructor needed? Can the existing public  constructor be used here (with `wrapWithExitableDirectoryReader` set to `false`)?